pkg/cache/sql/informer/factory/informer_factory.go

```diff
@@ -0,0 +1,134 @@
+/*
+Package factory provides an informer factory for the sql-based informer.
+*/
+package factory
+
+import (
+	"fmt"
+	"github.com/rancher/lasso/pkg/cache/sql/attachdriver"
+	db2 "github.com/rancher/lasso/pkg/cache/sql/db"
+	"github.com/rancher/lasso/pkg/cache/sql/encryption"
+	"github.com/rancher/lasso/pkg/cache/sql/informer"
+	sqlStore "github.com/rancher/lasso/pkg/cache/sql/store"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"os"
+	"sync"
+)
+
+// InformerFactory builds Informer instances and keeps a cache of instances it created
+type InformerFactory struct {
+	informerCreateLock sync.Mutex
+	wg                 wait.Group
+	dbClient           DBClient
+	stopCh             chan struct{}
+	encryptAll         bool
+
+	newInformer newInformer
+
+	cache map[schema.GroupVersionKind]*informer.Informer
+}
+
+type newInformer func(client dynamic.ResourceInterface, fields [][]string, gvk schema.GroupVersionKind, db sqlStore.DBClient, shouldEncrypt bool, namespace bool) (*informer.Informer, error)
+
+type DBClient interface {
+	informer.DBClient
+	sqlStore.DBClient
+	connector
+}
+
+type connector interface {
+	NewConnection() error
+}
+
+var defaultEncryptedResourceTypes = map[schema.GroupVersionKind]struct{}{
+	schema.GroupVersionKind{
+		Version: "v1",
+		Kind:    "Secret",
+	}: {},
+}
+
+const (
+	EncryptAllEnvVar = "CATTLE_ENCRYPT_CACHE_ALL"
+)
+
+func init() {
+	indexedFieldDBPath := db2.OnDiskInformerIndexedFieldDBPath
+	if os.Getenv(EncryptAllEnvVar) == "true" {
+		indexedFieldDBPath = ":memory:" // indexedFieldDBPath = "file::memory:?cache=shared"
+	}
+	attachdriver.Register("file:" + indexedFieldDBPath + "?cache=shared")
+}
+
+// NewInformerFactory returns an informer factory instance
+func NewInformerFactory() (*InformerFactory, error) {
+	m, err := encryption.NewManager()
+	if err != nil {
+		return nil, err
+	}
+	dbClient, err := db2.NewClient(nil, m, m)
+	if err != nil {
+		return nil, err
+	}
+	return &InformerFactory{
+		wg:          wait.Group{},
+		stopCh:      make(chan struct{}),
+		cache:       map[schema.GroupVersionKind]*informer.Informer{},
+		encryptAll:  os.Getenv(EncryptAllEnvVar) == "true",
```

Comment:
The agreement with security was for secrets and tokens to be encrypted by default with the option to toggle encryption for all. Pinging @macedogm for confirmation. This is also in doc and was signed off by security.

Reply:
> The agreement with security was for secrets and tokens to be encrypted by default with the option to toggle encryption for all.

Yes, that is the agreement, by default only sensitive fields are encrypted, for security and efficiency reasons, but the option to encrypt everything is provided.

If I understood correctly, [these](https://github.com/rancher/lasso/pull/65/files#diff-30ea1a028c2e55261cf0c7fd16267f209c53187a2fa4372afc2580001b9a668bR51-R56) are the default encrypted fields, right?

Question that I don't recollected the answer, where will these settings be documented?

Reply:
Tokens are not accessible via the v1 endpoint atm https://github.com/rancher/rancher/blob/release/v2.9/pkg/api/steve/disallow/disallow.go#L39.

Reply:
> where will these settings be documented?

It will all go to official docs, likely starting from an experimental paragraph in the 'Tuning and Best Practices for Rancher at Scale' section.

I plan to write that as soon as we have confirmation from QA that no big regressions exist with the option turned off.

Reply:
-> [epic](https://github.com/rancher/rancher/issues/45635)